### ISSUE TITLE
deprecate the nis module

### ISSUE THREAD

**Issue by ac224bc2-6200-4e59-bd22-a7f440782ac2** [python/cpython#76188]
BPO | [32007](https://bugs.python.org/issue32007)
--- | :---
Nosy | @tiran, @benjaminp, @mcepl, @floppym, @JelleZijlstra, @stratakis, @iritkatriel

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2022-03-13.19:01:35.147>
created_at = <Date 2017-11-11.17:06:22.963>
labels = ['type-feature', 'build', '3.11']
title = 'deprecate the nis module'
updated_at = <Date 2022-03-13.19:01:35.146>
user = 'https://github.com/floppym'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-13.19:01:35.146>
actor = 'iritkatriel'
assignee = 'none'
closed = True
closed_date = <Date 2022-03-13.19:01:35.147>
closer = 'iritkatriel'
components = ['Build']
creation = <Date 2017-11-11.17:06:22.963>
creator = 'floppymaster'
dependencies = []
files = []
hgrepos = []
issue_num = 32007
keywords = []
message_count = 7.0
messages = ['306090', '306138', '309699', '321074', '321173', '411342', '415078']
nosy_count = 7.0
nosy_names = ['christian.heimes', 'benjamin.peterson', 'mcepl', 'floppymaster', 'JelleZijlstra', 'cstratak', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = 'duplicate'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue32007'
versions = ['Python 3.11']
```

</p></details>

**Comment by ac224bc2-6200-4e59-bd22-a7f440782ac2** [comment]
The nis extension module fails to build against glibc-2.26 with the "obsolete-rpc" option disabled.

Downstream bug report: https://bugs.gentoo.org/631488

glibc-2.26 release notes: https://sourceware.org/ml/libc-alpha/2017-08/msg00010.html

The nis module is normally skipped since rpcsvc/yp_prot.h will not exist. However, if the external "libnsl" package is installed, compilation is attempted and fails as given below.

```c
building 'nis' extension
x86_64-pc-linux-gnu-gcc -pthread -fPIC -Wno-unused-result -Wsign-compare -DNDEBUG -march=native -O2 -pipe -g -fwrapv -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -I./Include -I. -I/var/tmp/portage/dev-lang/python-3.6.3/work/Python-3.6.3/Include -I/var/tmp/portage/dev-lang/python-3.6.3/work/Python-3.6.3 -c /var/tmp/portage/dev-lang/python-3.6.3/work/Python-3.6.3/Modules/nismodule.c -o build/temp.linux-x86_64-3.6/var/tmp/portage/dev-lang/python-3.6.3/work/Python-3.6.3/Modules/nismodule.o
/var/tmp/portage/dev-lang/python-3.6.3/work/Python-3.6.3/Modules/nismodule.c:17:21: fatal error: rpc/rpc.h: No such file or directory
 #include <rpc/rpc.h>
                     ^
compilation terminated.
```


To summarize:

glibc no longer provides rpc/rpc.h or rpcsvc/yp_prot.h. One must install a couple of external libraries to provide the same functionality.

The missing rpcsvc/yp_prot.h file can be resolved by installing the "libnsl" package, which installs the header where setup.py expects to find it.

The missing rpc/rpc.h file may be provided by the "libtirpc" package, but this installs the file under /usr/include/tirpc, which must be added to the include path (-I/usr/include/tirpc).

Ideally, pkg-config would be used to find both libnsl and libtirpc.

**Comment by benjaminp** [comment]
Rather than investing any effort in nis, we should probably just deprecate and eventually remove it.

(If glibc is actually getting rid of something, you know must be very very deprecated.)
